### ISSUE TITLE
Cannot find module %project_path%\node_modules\@tensorflow\tfjs-node\lib\napi-v5\tfjs_binding.node'

### ISSUE THREAD

**Issue by dnjooiopa** [tensorflow/tfjs#2678]
I import @tensorflow/tfjs-node in app.js and got this error when I run it.

#### TensorFlow.js version
@tensorflow/tfjs-node v1.5.1

#### Browser version
nodejs version

#### Code to reproduce the bug / link to feature request
```js
internal/modules/cjs/loader.js:800
    throw err;
    ^

Error: Cannot find module 'D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\lib\napi-v5\tfjs_binding.node'
Require stack:
- D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\dist\index.js
- D:\TESA\tfjs-test\part2\app.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:797:15)
    at Function.Module._load (internal/modules/cjs/loader.js:690:27)
    at Module.require (internal/modules/cjs/loader.js:852:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\dist\index.js:46:16)
    at Module._compile (internal/modules/cjs/loader.js:959:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:995:10)
    at Module.load (internal/modules/cjs/loader.js:815:32)
    at Function.Module._load (internal/modules/cjs/loader.js:727:14)
    at Module.require (internal/modules/cjs/loader.js:852:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    'D:\\TESA\\tfjs-test\\part2\\node_modules\\@tensorflow\\tfjs-node\\dist\\index.js',
    'D:\\TESA\\tfjs-test\\part2\\app.js'
  ]
}
```

Please anyone help me I'm new to tensorflowjs

**Comment by tafsiri** [comment]
@dnjooiopa Could you outline steps to reproduce this error, and tell us a bit more about how you set this up. Did you follow the instructions in https://www.tensorflow.org/js/tutorials/setup#nodejs_setup

**Comment by dnjooiopa** [comment]
I've followed the instructions in  https://www.tensorflow.org/js/tutorials/setup#nodejs_setup and got this error

in package.json file
```js
{
  "name": "part2",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "@tensorflow/tfjs-node": "^1.5.1"
  }
```

in app.js file

```js
const tf = require('@tensorflow/tfjs');

// Optional Load the binding:
// Use '@tensorflow/tfjs-node-gpu' if running with GPU.
require('@tensorflow/tfjs-node');

// Train a simple model:
const model = tf.sequential();
model.add(tf.layers.dense({units: 100, activation: 'relu', inputShape: [10]}));
model.add(tf.layers.dense({units: 1, activation: 'linear'}));
model.compile({optimizer: 'sgd', loss: 'meanSquaredError'});

const xs = tf.randomNormal([100, 10]);
const ys = tf.randomNormal([100, 1]);

model.fit(xs, ys, {
  epochs: 100,
  callbacks: {
    onEpochEnd: (epoch, log) => console.log(`Epoch ${epoch}: loss = ${log.loss}`)
  }
});
 ```

**Comment by kangyizhang** [comment]
@dnjooiopa 
1. can you check what is inside `D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\lib\napi-v5`?
2. try go to `D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node` and run `npm run build-addon-from-source`, paste the output and try if it works

**Comment by dnjooiopa** [comment]
1. In ```D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\lib\napi-v5 ``` there is a file named ```tensorflow.dll```

2. I try run ```npm run build-addon-from-source``` and got this new error
```js
> @tensorflow/tfjs-node@1.5.1 build-addon-from-source D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node
> node-pre-gyp install --build-from-source

node-pre-gyp WARN Using needle for node-pre-gyp https download 
gyp ERR! find VS 
gyp ERR! find VS msvs_version not set from command line or npm config
gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
gyp ERR! find VS could not use PowerShell to find Visual Studio 2017 or newer
gyp ERR! find VS looking for Visual Studio 2015
gyp ERR! find VS - not found
gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
gyp ERR! find VS
gyp ERR! find VS **************************************************************
gyp ERR! find VS You need to install the latest version of Visual Studio
gyp ERR! find VS including the "Desktop development with C++" workload.
gyp ERR! find VS For more information consult the documentation at:
gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
gyp ERR! find VS **************************************************************
gyp ERR! find VS 
gyp ERR! configure error
gyp ERR! stack Error: Could not find any Visual Studio installation to use
gyp ERR! stack     at VisualStudioFinder.fail (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:121:47)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:74:16
gyp ERR! stack     at VisualStudioFinder.findVisualStudio2013 (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:351:14)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:70:14
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\find-visualstudio.js:372:16
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\util.js:54:7
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\util.js:33:16
gyp ERR! stack     at ChildProcess.exithandler (child_process.js:302:5)
gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
gyp ERR! System Windows_NT 10.0.18363
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--build-from-source" "--module=D:\\TESA\\tfjs-test\\part2\\node_modules\\@tensorflow\\tfjs-node\\lib\\napi-v5\\tfjs_binding.node" "--module_name=tfjs_binding" "--module_path=D:\\TESA\\tfjs-test\\part2\\node_modules\\@tensorflow\\tfjs-node\\lib\\napi-v5" "--napi_version=5" "--node_abi_napi=napi" "--napi_build_version=5" "--node_napi_label=napi-v5"
gyp ERR! cwd D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node
gyp ERR! node -v v12.14.0
gyp ERR! node-gyp -v v5.0.5
gyp ERR! not ok 
node-pre-gyp ERR! build error
node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --build-from-source --module=D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\lib\napi-v5\tfjs_binding.node --module_name=tfjs_binding --module_path=D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\lib\napi-v5 --napi_version=5 --node_abi_napi=napi --napi_build_version=5 --node_napi_label=napi-v5' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (D:\TESA\tfjs-test\part2\node_modules\node-pre-gyp\lib\util\compile.js:83:29)
node-pre-gyp ERR! stack     at ChildProcess.emit (events.js:210:5)
node-pre-gyp ERR! stack     at maybeClose (internal/child_process.js:1021:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
node-pre-gyp ERR! System Windows_NT 10.0.18363
node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\TESA\\tfjs-test\\part2\\node_modules\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--build-from-source"
node-pre-gyp ERR! cwd D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node
node-pre-gyp ERR! node -v v12.14.0
node-pre-gyp ERR! node-pre-gyp -v v0.14.0
node-pre-gyp ERR! not ok
Failed to execute 'C:\Program Files\nodejs\node.exe C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --build-from-source --module=D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\lib\napi-v5\tfjs_binding.node --module_name=tfjs_binding --module_path=D:\TESA\tfjs-test\part2\node_modules\@tensorflow\tfjs-node\lib\napi-v5 --napi_version=5 --node_abi_napi=napi --napi_build_version=5 --node_napi_label=napi-v5' (1)
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @tensorflow/tfjs-node@1.5.1 build-addon-from-source: `node-pre-gyp install --build-from-source`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @tensorflow/tfjs-node@1.5.1 build-addon-from-source script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\Kittisak\AppData\Roaming\npm-cache\_logs\2020-01-17T03_39_18_828Z-debug.log
```

**Comment by kangyizhang** [comment]
@dnjooiopa can you try again with version 1.5.1? I just updated the addon module.

**Comment by dnjooiopa** [comment]
It works!!
Thank you for helping me.

**Comment by kangyizhang** [comment]
Glad to hear it works!

**Comment by martimpassos** [comment]
Hi, I'm getting the same error but ```tfjs_binding.node``` exists in ```tfjs-node/lib/napi-v5```, it seems to be missing the ```tensorflow.dll``` file mentioned above. I tried ```npm rebuild @tensorflow/tfjs-node --build-from-source``` without success, any leads?

**Comment by RezaRahmati** [comment]
I get same error using version 1.7.0 or 1.7.4

`!  Error: The specified module could not be found.
\\?\C:\projects\digital-verification\api\node_modules\@tensorflow\tfjs-node\lib\napi-v5\tfjs_binding.node
    at Object.Module._extensions..node (internal/modules/cjs/loader.js:1122:18)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (C:\projects\digital-verification\api\node_modules\@tensorflow\tfjs-node\dist\index.js:58:16)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
!  We were unable to load your functions code. (see above)
   - It appears your code is written in Typescript, which must be compiled before emulation.
[4:10:42 PM] Found 0 errors. Watching for file changes.`

I ran `npm run build-addon-from-source` in `node_modules/@tensorflow/tfjs-node` and still same error

**Comment by ramosisw** [comment]
> I get same error using version 1.7.0 or 1.7.4
> 
> `! Error: The specified module could not be found.
> \?\C:\projects\digital-verification\api\node_modules@tensorflow\tfjs-node\lib\napi-v5\tfjs_binding.node
> at Object.Module._extensions..node (internal/modules/cjs/loader.js:1122:18)
> at Module.load (internal/modules/cjs/loader.js:928:32)
> at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> at Module.require (internal/modules/cjs/loader.js:952:19)
> at require (internal/modules/cjs/helpers.js:88:18)
> at Object. (C:\projects\digital-verification\api\node_modules@tensorflow\tfjs-node\dist\index.js:58:16)
> at Module._compile (internal/modules/cjs/loader.js:1063:30)
> at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> at Module.load (internal/modules/cjs/loader.js:928:32)
> at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> ! We were unable to load your functions code. (see above)
> 
> * It appears your code is written in Typescript, which must be compiled before emulation.
>   [4:10:42 PM] Found 0 errors. Watching for file changes.`
> 
> I ran `npm run build-addon-from-source` in `node_modules/@tensorflow/tfjs-node` and still same error
> 
> ## Update
> moving `\node_modules\@tensorflow\tfjs-node\deps\lib\tensorflow.dll` to `\node_modules\@tensorflow\tfjs-node\lib\napi-v5\` solved the issue

SO: Windows 

This works for me, 
```sh
$ npm -v 
6.14.11

$ npm list | Select-String -Pattern tensorflow
+-- @tensorflow/tfjs-node@1.7.4
| +-- @tensorflow/tfjs@1.7.4
| | +-- @tensorflow/tfjs-converter@1.7.4
| | +-- @tensorflow/tfjs-core@1.7.4 deduped
| | +-- @tensorflow/tfjs-data@1.7.4
| | `-- @tensorflow/tfjs-layers@1.7.4
| +-- @tensorflow/tfjs-core@1.7.4
```

**Comment by alkaaf** [comment]
> I get same error using version 1.7.0 or 1.7.4
> 
> `! Error: The specified module could not be found.
> \?\C:\projects\digital-verification\api\node_modules@tensorflow\tfjs-node\lib\napi-v5\tfjs_binding.node
> at Object.Module._extensions..node (internal/modules/cjs/loader.js:1122:18)
> at Module.load (internal/modules/cjs/loader.js:928:32)
> at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> at Module.require (internal/modules/cjs/loader.js:952:19)
> at require (internal/modules/cjs/helpers.js:88:18)
> at Object. (C:\projects\digital-verification\api\node_modules@tensorflow\tfjs-node\dist\index.js:58:16)
> at Module._compile (internal/modules/cjs/loader.js:1063:30)
> at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
> at Module.load (internal/modules/cjs/loader.js:928:32)
> at Function.Module._load (internal/modules/cjs/loader.js:769:14)
> ! We were unable to load your functions code. (see above)
> 
> * It appears your code is written in Typescript, which must be compiled before emulation.
>   [4:10:42 PM] Found 0 errors. Watching for file changes.`
> 
> I ran `npm run build-addon-from-source` in `node_modules/@tensorflow/tfjs-node` and still same error
> 
> ## Update
> moving `\node_modules\@tensorflow\tfjs-node\deps\lib\tensorflow.dll` to `\node_modules\@tensorflow\tfjs-node\lib\napi-v5\` solved the issue

yep. moving tensorflow.dll is working for me.

**Comment by juancruzsosagab** [comment]
Same for me! But i've got anothe issue
node_modules\@tensorflow\tfjs-core\dist\backends\backend.js:515
    throw new Error("'" + kernelName + "' not yet implemented or not found in the registry. " +
          ^

Error: 'softmax' not yet implemented or not found in the registry. Did you forget to import the kernel?
